### PR TITLE
Feature/#25 ui for keywords

### PR DESCRIPTION
**Pull request by oyvi96**

### CODE REVIEW

File: frontend/database/general.py [hunk 1]
```diff
@@ -168,6 +174,80 @@ def getCategoryNames():
     return [[x[0], x[1]] for x in data]
 
 
+def get_keywords_from_titles(titles, quantity=1000):
```
Comment (kbalog) [inline frontend/database/general.py]: rename variable `quantity` to `num`

File: frontend/database/general.py [hunk 2]
```diff
@@ -168,6 +174,80 @@ def getCategoryNames():
...
+    """Returns a list of keywords for a list of scientific paper titles.
+    Can also specify quantity of keywords to return."""
```
Comment (kbalog) [inline frontend/database/general.py]: "...specify the number of ..."

File: frontend/database/general.py [hunk 3]
```diff
@@ -168,6 +174,80 @@ def getCategoryNames():
...
+        else:
+            row = row[0]
+        if row['opinion'] == 'discarded':
+            continue
+        keywords[row['keyword']] += row['score']
+
+    return keywords
+
+
+def store_keyword_opinion(userid, keyword, opinion):
```
Comment (kbalog) [inline frontend/database/general.py]: "opinion" => "feedback" everywhere

File: frontend/views/general.py [hunk 4]
```diff
@@ -173,6 +174,26 @@ def registerSystemPage():
...
+    Returns list of keywords or an empty string for failure."""
+    try:
+        author_titles = find_author_titles(author_url)
+        keywords = db.get_keywords_from_titles(author_titles)
+    except ValueError as e:
+        return jsonify(error=str(e))
+    return jsonify(keywords=keywords)
+
+@mod.route('/keyword_opinion/<keyword>/<opinion>', methods=['GET'])
+def send_user_opinion(keyword, opinion):
```
Comment (kbalog) [inline frontend/views/general.py]: "opinion" => "feedback"

File: frontend/README.md [hunk 5]
```diff
@@ -58,6 +58,14 @@ Routes available:
 
 >Likes or unlikes articles from web, depending on current state. Returns "Success" or "Fail".
 
+``/author_keywords/<author_url> [GET]``
+
+>Scrapes the url provided for paper titles and looks up titles in database to find matching keywords. Returns "" or list of keywords.
+
+``/keyword_opinion/<keyword>/<opinion> [GET]``
```
[the remark this answers is not in the document]
Comment (kgingstad) [reply frontend/README.md]: renamed

File: frontend/database/general.py (hunk 1 above)
Comment (kgingstad) [reply frontend/database/general.py]: renamed

**Pull request merged**